src/mainwindow.cpp

```diff
@@ -245,5 +236,43 @@ void MainWindow::on_actionWEST_triggered()
 
 void MainWindow::setNumberOfSteps(QString& c)
 {
-    countNumberOfSteps->setText(c);
+	statusbarItems.stepsValue->setText(c);
+}
+
+void MainWindow::StatusbarItems::init(QStatusBar* statusbar) {
+	stepsKey = new QLabel(statusbar); // parent takes ownership
+	stepsValue = new QLabel(statusbar); // parent takes ownership
+
```

Comment:
Here you'd have to add the initialisation code for additional labels or other widgets inside status bar.